src/vaults/PoolManager.sol

```diff
@@ -369,29 +369,33 @@ contract PoolManager is
 
     /// @inheritdoc IPoolManager
     function linkVault(PoolId poolId, ShareClassId scId, AssetId assetId, IBaseVault vault) public auth {
-        _shareClass(poolId, scId);
+        ShareClassDetails storage shareClass = _shareClass(poolId, scId);
 
         AssetIdKey memory assetIdKey = _idToAsset[assetId];
 
         IBaseRequestManager manager = vault.manager();
         manager.addVault(poolId, scId, vault, assetIdKey.asset, assetId);
-
         _vaultDetails[vault].isLinked = true;
 
+        IAuth(address(shareClass.shareToken)).rely(address(vault));
```

Comment:
Unless I miss something, I believe it is a lot cleaner to set this up here in the pool manager, as this removes the vault managers as wards from the share tokens entirely.